examples/nodejs/get-set-batch-perf-test/perf-test.ts

```diff
@@ -0,0 +1,251 @@
+import {
+  CacheClient,
+  CacheFlush,
+  CacheSet,
+  DefaultMomentoLoggerFactory,
+  DefaultMomentoLoggerLevel,
+  MomentoLogger,
+  MomentoLoggerFactory,
+} from '@gomomento/sdk';
+import {createCache, getCacheClient} from './utils/cache';
+import {GetSetConfig, PerfTestConfiguration, PerfTestOptions, RequestType} from './utils/perf-test-options';
+import {calculateSummary, getElapsedMillis} from './utils/perf-test-utils';
+
+class PerfTest {
+  private readonly momentoClientPromise: Promise<CacheClient>;
+  private momento: CacheClient;
+  private readonly loggerFactory: MomentoLoggerFactory;
+  private readonly logger: MomentoLogger;
+  private readonly cacheItemTtlSeconds = 600;
+  private readonly options: PerfTestOptions;
+  private readonly testConfiguration: PerfTestConfiguration;
+  private readonly cacheName: string = 'js-perf-test';
+  private totalRequests = 0;
```

Comment:
for these variables that are tracking state from an individual run, you should put them into a single `Context` object (you can define it as either a class or an interface) so that you don't have to worry about resetting all of them explicitly between each run; just create a new context object.

Also, you should always use a histogram when you are writing perf test code; it will allow you to get rid of your completion times array (which might grow *very* large if you are making millions of requests).

Reply:
we will want to make sure we're recording percentile info from the histogram as part of your output. You can probably steal some of the histogram output code from the other loadgen if you like.

Reply:
We will still need to reset the context in between two set/get configs. For example for Set - we need to reset in between `{batchSize: 5, itemSizeBytes: 500}` and `{batchSize: 5, itemSizeBytes: 1000}`.